src/include/distributed/recursive_planning.h

```diff
@@ -22,6 +22,32 @@
 #include "nodes/relation.h"
 #endif
 
+/* managed via guc.c */
+typedef enum
+{
+	LOCAL_JOIN_POLICY_NEVER = 0,
+	LOCAL_JOIN_POLICY_PREFER_LOCAL = 1,
+	LOCAL_JOIN_POLICY_PREFER_DISTRIBUTED = 2,
+	LOCAL_JOIN_POLICY_AUTO = 3,
+} LocalJoinPolicy;
+
+extern int LocalTableJoinPolicy;
+
+/*
+ * RecursivePlanningContext is used to recursively plan subqueries
+ * and CTEs, pull results to the coordinator, and push it back into
+ * the workers.
+ */
+typedef struct RecursivePlanningContext
```

Comment:
I guess the only piece that might be a problem is calling `ReplaceRTERelationWithRteSubquery`. Because it expects `RecursivePlanningContext` as it calls `RecursivelyPlanSubquery`. Other places can be replaced, but not sure if we can get rid of this